### PR TITLE
feature/new-youtube-api-update

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** #11 

**This PR will result in the following new package version:** `v0.4.0`

This will be breaking due to the removal of the `video_metadata` content.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

### 🚨 Breaking Changes 🚨:
- Removed support for the `video_metadata` Cloud Function generated source table and downstream models. This also means the following variables are no longer used or necessary within the package:
  - `youtube__using_video_metadata`
  - `youtube_metadata_schema`
  - `youtube_analytics_database`
- To be consistent with our other packages, the identifier variables have been updated. Please see the following changes to the identifier variables used in this package.

| **old identifier name** | **new identifier name** |
| ------------------------|-------------------------|
| `youtube__channel_basic_table` | `youtube_analytics_channel_basic_a_2_identifier` |
| `youtube__channel_demographics_table` | `youtube_analytics_channel_demographics_a_1_identifier` |

### Feature Updates
- Following the latest update to **all** Fivetran YouTube Analytics connectors, the `video` metadata is now synced as part of the connector by default. Therefore, the `video` source along with the following staging models have been added:
  - `stg_youtube__video_tmp`
  - `stg_youtube__video`

### Under the Hood:
- Incorporated the new `fivetran_utils.drop_schemas_automation` macro into the end of each Buildkite integration test job. ([#10](https://github.com/fivetran/dbt_youtube_analytics_source/pull/10))
- Updated the pull request [templates](/.github). ([#10](https://github.com/fivetran/dbt_youtube_analytics_source/pull/10))
- Included auto-releaser GitHub Actions workflow to automate future releases

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [X] dbt run –full-refresh && dbt test
- [X] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [X] The appropriate issue has been linked, tagged, and properly assigned.
- [X] All necessary documentation and version upgrades have been applied. Don't forget to update the version in the README if necessary.
- [X] docs were regenerated (unless this PR does not include any code or yml updates).
- [ ] BuildKite integration tests are passing.
- [x] Detailed validation steps have been provided below.

### Detailed Validation
Please share any and all of your validation steps:

Please view the corresponding Height ticket for all the validations performed in order to gain comfort over these changes. The following validations are included in the Hex Validations notebook:
- Confirmation of the new `stg_youtube__video` model
- Validation that the same type of information is being provided in the `stg_youtube__video` model that was previously present in the `stg_youtube__video_metadata` model.

### If you had to summarize this PR in an emoji, which would it be?
 
📹

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,20 +77,16 @@ models:
...
-
 vars:
-  youtube__channel_basic_table:         "my_channel_basic_table_name"
-  youtube__channel_demographics_table:  "demographics_youtube_report"
+    youtube_analytics_<default_source_table_name>_identifier: your_table_name 
 ```
 
-</details>
-
 ## (Optional) Step 8: Orchestrate your models with Fivetran Transformations for dbt Core™
```
Comment (fivetran-avinash) [inline README.md]: Should be Step 6 now

File: models/stg_youtube.yml [hunk 2]
```diff
@@ -68,30 +67,94 @@ models:
...
+        description: Boolean indicating if the video has a custom thumbnail
+      - name: content_details_licensed_content
+        description: Boolean indicating if the video has licensed content
+      - name: content_details_projection
+        description: Identifies the projection type of the video
+      - name: content_details_region_restriction
+        description: Identifies the region restrictions of the video
+      - name: etag
+        description: Electronic tag of the video
       - name: video_id
```
Comment (fivetran-avinash) [inline models/stg_youtube.yml]: would it make sense to move the grain field to the topmost column for easier readability.

**Review by fivetran-avinash — APPROVED** [review]
Validated the data looks all good! One small README tweak and one suggestion but this should be good to go.

File: README.md (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply README.md]: Good catch! Updated

File: models/stg_youtube.yml (hunk 2 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_youtube.yml]: Makes sense to bring it to the top for easier readability. Just updated. Thanks!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,8 +1,27 @@
-# dbt_youtube_analytics_source v0.UPDATE.UPDATE
+# dbt_youtube_analytics_source v0.4.0
```
Comment (fivetran-catfritz) [inline CHANGELOG.md:2]: Just [the same comment as the transform](https://github.com/fivetran/dbt_youtube_analytics/pull/13#discussion_r1424225513) here.

**Review by fivetran-catfritz — CHANGES_REQUESTED** [review]
Same small suggestion for the changelog. 🙂

File: CHANGELOG.md (hunk 3 above)
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md:2]: Thanks, just updated

**Review by fivetran-catfritz — APPROVED** [review]
lgtm!

**Pull request merged**